### PR TITLE
clang-cl: add a translation pass for `-isystem` args to work

### PR DESCRIPTION
**Pull request by Sahnvour**
Related to #7114.

I'm not sure how to please the type checker on this CI failure.

### CODE REVIEW

File: mesonbuild/compilers/mixins/visualstudio.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 if T.TYPE_CHECKING:
     from ...environment import Environment
+    from ..dependencies import Dependency
```
Comment (dcbaker) [inline mesonbuild/compilers/mixins/visualstudio.py:29]: this should be:
```python
from ...dependencies import Dependency
```

That will make the type checker happy I think.

**Review by dcbaker — CHANGES_REQUESTED** [review]
I think with the import resolved the type checker will be happy. I haven't had a chance to get windows out and test this yet, and I haven't looked to see if we have tests that should now run with clang-cl that skip

File: mesonbuild/compilers/mixins/visualstudio.py (hunk 1 above)
Comment (Sahnvour) [reply mesonbuild/compilers/mixins/visualstudio.py]: Right, thanks.

**Comment by Sahnvour** [comment]
This should be good to go, despite the spurious failures.

**Review by dcbaker — APPROVED** [review]

**Review by dcbaker — APPROVED** [review]

**Comment by Sahnvour** [comment]
Failures are related to Codecov.
Thanks for the approval @dcbaker, any chance this could be merged in the near future ? That would really help having cleaner code.

**Comment by eli-schwartz** [comment]
> Failures are related to Codecov.

Codecov should never be able to fail. It's just run as a post-test thing to upload coverage results for both successful and failing CI.

**Pull request merged**